### ISSUE TITLE
Cypress Runner - New Feature: Default Folders as Closed

### ISSUE THREAD

**Issue by zskelton** [cypress-io/cypress#26141]
### What would you like?

1. In Cypress Runner, the tests display always with the folder structure open. 
Requested Feature: Setting in the config file to allow folders to default to closed.

2. Each Folder in the list is either opened or closed individually.
Requested Feature: A + and - icon to open/close folders as a group.

### Why is this needed?

Allows for visibility of tests in runner to be specific to the group of tests being debugged.

## How it Looks Now:
![image_1_normal](https://user-images.githubusercontent.com/643992/225831456-eab18f04-0ba0-46df-b995-09e12402a128.png)

## What Could Be Added:
![image_2_concept](https://user-images.githubusercontent.com/643992/225831459-15ebc729-b01d-4003-833f-dca8b453d44e.png)

**Comment by lmiller1990** [comment]
VS Code has a similar feature:

<img width="272" alt="image" src="https://user-images.githubusercontent.com/19196536/226252020-3232976f-9f4c-4bba-9c0c-4430ea986ba9.png">

I agree, our file explorer could definitely do with some improvements. I'lll mark this as a feature request. Is this something you'd be interested in implementing, if we decide to move forward with it?

**Comment by zskelton** [comment]
Yep, my friend and I were digging into the source last weekend so I think this would be a good issue to contrib on.

Just curious, is there something missing for the build on Windows? I spent hours trying to get the build script working on Windows  and got plenty of build errors and never quite got working binary/tests. However, I pulled my fork into my WSL Linux and it built and ran like a charm.

Not really a problem as WSL/VS Code works great, but was curious if I was missing a step or there was a known issue in the build script.

Thanks!
________________________________

**Comment by lmiller1990** [comment]
I have done development on windows using git bash at my shell, but only to debug windows specific things... I don't know if anyone at Cypress is doing development on windows on a regular basis, so it is entirely possible something isn't working. If you can share the issue, I can take a look - but to my knowledge, it should work just fine. It should just be `yarn` to install the deps and `yarn dev` to get started in watch mode. This is how I develop for Cypress.

As for actually *building* a binary - I haven't done this on windows, it's definitely working (we release for all platforms every 2 weeks) but I haven't actually built locally on windows before. You shouldn't need to, though - just start the app in watch mode and off you go.

I will need to get some advice/guidance from our design team, I'd recommend waiting until we have something from them, as not to waste your time.

Also, the part of the code that builds the spec list is pretty gnarly, but we have an experimental feature - [run all specs](https://www.cypress.io/blog/2023/01/12/check-out-our-experimental-release-of-run-all-specs/) that adds a folder level behavior (see the article and GIF) so you might be able to use that code path to implement this.

Please give me a few days to see what the design team thinks is the best way to move forward with this feature request.

**Comment by zskelton** [comment]
No problem, just let me know....
________________________________

**Comment by lmiller1990** [comment]
I took another look at VS Code. It looks like they do not have a similar feature on a directory-by-directory basis, only a button to do all collapsing/uncollapsing at once.

![image](https://user-images.githubusercontent.com/19196536/227386521-c10e7bb8-8c1e-431a-a997-9be381fbeabf.png)

I talked to our design team - a change to the UI has to be carefully considered and weighed up, and this would add quite a lot of extra icons to the UI, and be something that is widely requested.

I'm not sure we will be moving forward with this feature request in the immediate future. For now, we can wait and see if there is more desire for this feature. If more users express a need for this, we can consider it.

**Comment by zskelton** [comment]
Not a problem, just let me know if the status changes!
________________________________

**Comment by lmiller1990** [comment]
@zskelton just another question - is there a particular use case you've got? Do you have a huge amount of specs (like 500+)? Are those E2E or Component specs?

Knowing the use cases can help better prioritize and move feature requests to something we can work on.

**Comment by zskelton** [comment]
No particular use case, just a thought 'wouldn't this be nice'. Right now doing in component tests and thought it would have a cleaner look and would be a contrib opportunity.
________________________________

**Comment by lmiller1990** [comment]
Thanks for the feedback; we track all the feedback, once something is requested a few times, and we've got a good idea of the us cases, we will look into it more - for now it's just a nice to have, so I'll leave this one open to get more feedback before we commit to moving forward with any changes.

As an aside, if you've got any feedback around component testing, please let me know :100:

**Comment by DamirPecnik** [comment]
I propose that we configure this in the config file so that when we initiate the cypress open command, triggering our End-to-End (E2E) testing procedure, all folders are automatically set to **close**. if someone has over 20 folders and over 30 subfolders its hard to turn all of them off.

**Comment by kriedel-ifm** [comment]
Any update on this feature request? I'm encountering a similar need in my project:

Current situation:
- Large number of folders and sub-folders in the test structure
- Need to scroll extensively each time the runner opens or a test starts
- Manual folder collapsing required on every test run/session

Feature request:
1. Option to have all folders closed by default (via configuration)
2. Remember user's folder open/close state between sessions

This would significantly improve the user experience and efficiency when working with large test suites. It aligns with Cypress's goal of providing a smooth testing experience.

Has there been any consideration for implementing this? If not, I'd be happy to provide more details on use cases or contribute to the discussion on potential implementation approaches.

**Comment by trav-giv** [comment]
I want to chime in here and add my voice to the mix. This would be quite helpful as it makes navigating through directories and subdirectories much easier. I don't remember the exact name of all of my spec files, so searching isn't always quite right. As my test repository grows, this will become more and more useful and seems like it is probably a small lift? I think @DamirPecnik has the right idea:

> I propose that we configure this in the config file so that when we initiate the cypress open command, triggering our End-to-End (E2E) testing procedure, all folders are automatically set to close. if someone has over 20 folders and over 30 subfolders its hard to turn all of them off.

I was surprised to find this wasn't already a config option.
